daffodil-cli/src/it/scala/org/apache/daffodil/blob/TestBlob.scala

```diff
@@ -0,0 +1,191 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.blob
+
+import junit.framework.Assert._
+import org.junit.Test
+import java.io.File
+import org.apache.daffodil.CLI.Util
+import net.sf.expectit.matcher.Matchers.contains
+import net.sf.expectit.matcher.Matchers.eof
+
+class TestBlob {
+
+  val output1 = Util.getExpectedString("output1.txt")
+  val output1_nopretty = Util.getExpectedString("output1_nopretty.txt")
+  val output2 = Util.getExpectedString("output2.txt")
+  val output4 = Util.getExpectedString("output4.txt")
+  val output6 = Util.getExpectedString("output6.txt")
+  val output8 = Util.getExpectedString("output8.txt")
+  val output9 = Util.getExpectedString("output9.txt")
+  val output10 = Util.getExpectedString("output10.txt")
+  val output12 = Util.getExpectedString("output12.txt")
+
+  // Call md5sum with two different files and return if they match
+  /*def md5sum(shell: Expect, a: String, b: String): Boolean {
+    val cmdA = String.format("/usr/bin/md5sum %s | awk '{print $1}'", a)
+    shell.sendLine(cmd)
+    val resA = shell.expect("\n").toString
+
+    val cmdB = String.format("/usr/bin/md5sum %s | awk '{print $1}'", b)
+    shell.sendLine(cmd)
+    val resB = shell.expect("\n").toString
+
+    return resA == resB
+  }*/
+
+  @Test def test_basic_blob() {
+
+    val schemaFile = Util.daffodilPath("daffodil-cli/src/it/resources/org/apache/daffodil/CLI/blob.dfdl.xsd")
+    val inputFile = Util.daffodilPath("daffodil-cli/src/it/resources/org/apache/daffodil/CLI/input/blob_01.bin")
+    val (testSchemaFile, testInputFile) = if (Util.isWindows) (Util.cmdConvert(schemaFile), Util.cmdConvert(inputFile)) else (schemaFile, inputFile)
+
+    val shell = Util.start("")
+
+    try {
+      val cmd = String.format("%s parse -s %s %s", Util.binPath, testSchemaFile, testInputFile)
+      shell.sendLine(cmd)
+      val result = shell.expect(contains("tns:blob_01")).toString
+      // Use +7 to drop the 'file://' to get the path of the file
+      val generated_blob = result.substring(result.indexOf("file://") + 7).takeWhile(_ != '<')
+
+      //val checksum = String.format("%s parse -s %s %s", Util.binPath, testSchemaFile, testInputFile)
+      val cmp = String.format("/usr/bin/cmp %s %s", generated_blob, testInputFile)
```

Comment:
This is going to fail on windows. Maybe refactor the logic in XMLUtils used for TDML comparing blobs so that we can use it here as well? Not everything in a CLI tests has to be done via shell.send/expect.